diff --git a/node_modules/@aws-sdk/ec2-metadata-service/dist-cjs/index.js b/node_modules/@aws-sdk/ec2-metadata-service/dist-cjs/index.js
index 348a134..4be6050 100644
--- a/node_modules/@aws-sdk/ec2-metadata-service/dist-cjs/index.js
+++ b/node_modules/@aws-sdk/ec2-metadata-service/dist-cjs/index.js
@@ -116,6 +116,10 @@ class MetadataService {
                 return utilStream.sdkStreamMixin(response.body).transformToString();
             }
             else {
+                if (response.body) {
+                    // Consume body
+                    utilStream.sdkStreamMixin(response.body).transformToString();
+                }
                 throw new Error(`Failed to fetch metadata token with status code ${response.statusCode}`);
             }
         }
